### PR TITLE
Lint fixes / Demo0

### PR DESCRIPTION
**Pull request by nkbt**
One-by one syntax/lint fixes...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
   "scripts": {
     "start": "NODE_ENV=development node server.js",
     "build": "NODE_ENV=production webpack -p",
-    "lint": "eslint .",
+    "lint": "eslint --ext .js,.jsx .",
```
Comment (nkbt) [inline package.json]: Completely forgot we have jsx here...

**Comment by chenglou** [comment]
you can just fix all the demos at once I think

**Comment by chenglou** [comment]
Also the triple equal rule is too strong. I use `a == null` to captured undefined too and that's the only place I use it.

**Comment by nkbt** [comment]
Hm. Double equal is quite unsafe. It is better to be more explicit. If you want to capture undefined AND null, then it should be actully written like that =).

Will fix all demos, ok

**Comment by chenglou** [comment]
Kinda used to it and Facebook internally allows it just fine. Don't worry, won't abuse it here.

**Comment by nkbt** [comment]
Okay. Will update eslint config then.

File: .eslintrc [hunk 2]
```diff
@@ -8,6 +8,10 @@
   ],
   "rules": {
     "comma-dangle": [2, "always-multiline"],
+    "no-eq-null": 0,
+    "prefer-const": 2,
```
Comment (nkbt) [inline .eslintrc]: need to make pr to airbnb =)

File: Spring.js [hunk 3]
```diff
@@ -140,7 +140,7 @@ function noSpeed(coll) {
 }
 
 
-export default React.createClass({
+const Spring = React.createClass({
```
Comment (nkbt) [inline Spring.js]: We should have var so react can find out the component name.

File: utils.js [hunk 4]
```diff
@@ -1,5 +1,3 @@
-'use strict';
-
 // damn it JS
 export function clone(coll) {
   return JSON.parse(JSON.stringify(coll));
```
Comment (nkbt) [inline utils.js]: Better not to do this thing. Horrible performance.

Comment (chenglou) [inline utils.js]: Oh I'm very aware of that =) It probably prevented all sort of bugs during iteration for me. I have a todo to remove those once I'm done.

Comment (chenglou) [inline utils.js]: We also allocate like crazy in Spring. All these will be refactored once things settle down, so don't worry about it for now.

**Comment by chenglou** [comment]
(Is this done?)

**Comment by nkbt** [comment]
Yep. Walked through all modules and lint report is nice and clean now.

File: utils.js (hunk 4 above)
Comment (nkbt) [inline utils.js]: Yeah, didn't touch it now of course.

File: demo3/Demo.jsx [hunk 5]
```diff
@@ -148,12 +149,12 @@ let Demo = React.createClass({
                           type="checkbox"
                           onChange={this.handleDone.bind(null, date)}
                           checked={isDone}
-                        />
+                          />
```
Comment (chenglou) [inline demo3/Demo.jsx]: Is this a lint rule? Old one's fine?

Comment (nkbt) [inline demo3/Demo.jsx]: Nah... I think it is my autoformatter. Do you think it should not be all aligned?

Comment (chenglou) [inline demo3/Demo.jsx]: Always been coding in the old style (fb's internal style)

Comment (nkbt) [inline demo3/Demo.jsx]: Btw I usually keep closing bracket on the last line, so it does not look awkward.
Like:

```js
                        <button
                          className="destroy"
                          onClick={this.handleDestroy.bind(null, date)} />
```

Comment (chenglou) [inline demo3/Demo.jsx]: Same, but got used to the internal convention

Comment (nkbt) [inline demo3/Demo.jsx]: ok. reverted that change.

**Comment by chenglou** [comment]
Merging! Btw the style nits really have nothing to do with whatever "internal rules". Just got used to them personally and didn't want to juggle between them =)

**Pull request merged**